src/main/scala/org/apache/spark/shuffle/rss/BlockDownloaderPartitionRecordIterator.scala

```diff
@@ -164,12 +170,20 @@ class BlockDownloaderPartitionRecordIterator[K, C](
     val compressedLen = ByteBufUtils.readInt(bytes, 0)
     val uncompressedLen = ByteBufUtils.readInt(bytes, Integer.BYTES)
     val uncompressedBytes = new Array[Byte](uncompressedLen)
-    val count = decompressor.decompress(bytes, Integer.BYTES + Integer.BYTES, uncompressedBytes, 0, uncompressedLen)
-    decompressTime += (System.nanoTime() - decompressStartTime)
-    if (count != compressedLen) {
-      throw new RssInvalidDataException(
-        s"Data corrupted for shuffle $shuffleId partition $partition, expected compressed length: $compressedLen, but it is: $count, " + String.valueOf(downloader))
+    if (Compression.COMPRESSION_CODEC_ZSTD.equals(decompression)) {
+      val n = Zstd.decompress(uncompressedBytes, bytes)
```

Comment:
I think the bytes also inclued uncompressedLen and compressedLen. Better to use `Zstd.decompressByteArray(uncompressedBytes, 0, uncompressedLen, bytes, byteOffset, bytesize - byteOffset)` ? (byteOffset=Integer.BYTES + Integer.BYTES)

Reply:
Good catch, you are right! But I did not find method on Zstd to decompress source byte array with given offset. Where did you find method Zstd.decompressByteArray?

Reply:
https://github.com/luben/zstd-jni/blob/master/src/main/java/com/github/luben/zstd/Zstd.java#L406 I saw it from here.

Reply:
Got it, looks like Spark 2.4 uses an old version of Zstd which does not have `Zstd.decompressByteArray` method. I updated the code here a little bit to copy data in another byte array. Could optimize that for higher version of Spark using `Zstd.decompressByteArray` later.

Reply:
great, thank you!!